pdns/iputils.cc

```diff
@@ -54,19 +54,37 @@ int SSocket(int family, int type, int flags)
   return ret;
 }
 
-int SConnect(int sockfd, const ComboAddress& remote)
+int doConnect(int sockfd, [[maybe_unused]] bool fastopen, const ComboAddress& remote)
 {
-  int ret = connect(sockfd, reinterpret_cast<const struct sockaddr*>(&remote), remote.getSocklen()); // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
+#ifdef CONNECTX_FASTOPEN
```

Comment:
I see in the man page for `connectx` that `connectx() is currently supported only on AF_INET and AF_INET6 sockets of type SOCK_DGRAM and SOCK_STREAM`, which worries me a bit. The documentation does not seem to state what happens in that case, but I was wondering if it might make sense to fallback to regular `connect` when `connectx` fails?

Reply:
There is some assertion fail here, but honestly I don't know what that does (it seems it can be either noop or some sort of failure)

https://github.com/apple-oss-distributions/xnu/blob/e3723e1f17661b24996789d8afc084c0c3303b26/bsd/netinet/tcp_usrreq.c#L735

It might be a good idea to fallback to regular connect, yes